### PR TITLE
[fpv/csr] Enable checking CSRs with regwen

### PR DESCRIPTION
**Pull request by cindychip**
This PR enables formal to check CSRs regs which could be gated by
regwen.
The regwen can be HRO or HRW, so we need to probe to see the regwen
value. Because DV and FPV testbench has a slightly different path, we
use `FPV_ON` macro to distinguish which path to use.

Another commit is related to binding the CSR assertions to top-level.
We do not want to bind CSR assertions to top-level to avoid creating long
simulation/compile time. So I added a macro define.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

**Review by weicaiyang — APPROVED** [review]
LGTM

**Review by msfschaffner — APPROVED** [review]
LGTM

File: hw/top_earlgrey/dv/chip_sim_cfg.hjson [hunk 1]
```diff
@@ -103,6 +103,9 @@
   uvm_test_seq: chip_sw_base_vseq
   sw_build_device: sim_dv
 
+  // Add a default build option to indicate it is a top-level DV testbench.
+  build_opts: ["+define+TOP_LEVEL_DV=1"]
```
Comment (weicaiyang) [inline hw/top_earlgrey/dv/chip_sim_cfg.hjson:107]: nit: assigning a value seems not needed
```suggestion
  build_opts: ["+define+TOP_LEVEL_DV"]
```

**Review by weicaiyang — APPROVED** [review]
LGTM with nit

File: hw/top_earlgrey/dv/chip_sim_cfg.hjson (hunk 1 above)
Comment (cindychip) [reply hw/top_earlgrey/dv/chip_sim_cfg.hjson]: Thanks! Updated

**Pull request merged**